scripts/build.frontend.sh

```diff
@@ -1,11 +1,22 @@
 #!/bin/bash
+# Assumes it is run from the project root.
 
-# Ubuntu / Debian
-#   curl -sL https://deb.nodesource.com/setup_14.x | sudo -E bash -
-#   sudo apt-get install -y nodejs
-#
-# MacOS
-#   sudo port install nodejs14 npm6
+set -e
+
+if [ ! -d "_frontend" ]; then
+    echo "Checking out  submodules..."
+    git submodule update --init --recursive
+fi
+
+# Build within a Node container
```

Comment:
does this mean when we build.frontend.sh on our dev machines it will now also be done in docker?

Reply:
Yes. Though, you could bypass it if you know to use the `--exec` option.

Reply:
ok, thanks.   just wondering about what the changes here imply to machines where i dont have docker setup.  (e.g. i have some azure machines i just build and run houston/frontend that are blissfully clueless of docker.)

Reply:
This is the right call, we should expect Docker to be installed and not a specific NPM version.   Let's add a section to the README to say that Docker is required to build the frontend.